hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -580,3 +794,334 @@ def _dict_update(dest, upd, recursive_update=True, merge_lists=False):
             for k in upd:
                 dest[k] = upd[k]
         return dest
+
+
+def _osqueryd_running_status(pidfile, servicename):
+    '''
+    This function will check whether osqueryd is running in *nix systems
+    '''
+    osqueryd_running = False
+    if os.path.isfile(pidfile):
+      try:
+        with open(pidfile, 'r') as f:
+          xpid = f.readline().strip()
+          try:
+            xpid = int(xpid)
+          except:
+            xpid = 0
+            log.warn('unable to parse pid="{pid}" in pidfile={file}'.format(pid=xpid,file=pidfile))
+          if xpid:
+            log.info('pidfile={file} exists and contains pid={pid}'.format(file=pidfile, pid=xpid))
+            if os.path.isdir("/proc/{pid}".format(pid=xpid)):
+              try:
+                with open("/proc/{pid}/cmdline".format(pid=xpid),'r') as f2:
+                  cmdline = f2.readline().strip().strip('\x00').replace('\x00',' ')
+                  if 'osqueryd' in cmdline:
+                    log.info("process folder present and process is osqueryd")
+                    osqueryd_running = True
+                  else:
+                    log.error("process is not osqueryd, attempting to start osqueryd")
+              except:
+                log.error("process's cmdline cannot be determined, attempting to start osqueryd")
+            else:
+              log.error("process folder not present, attempting to start osqueryd")
+          else:
+            log.error("pid cannot be determined, attempting to start osqueryd")
+      except:
+        log.error("unable to open pidfile, attempting to start osqueryd")
+    else:
+      cmd = ['pkill', 'osqueryd']
```

Comment:
pkill osqueryd might kill osquery daemon instances started by user outside/independent of Hubble. Is it the only way to kill osquery daemon ? Can process name be tweaked to something like 'hubble_osqueryd' and only pkill hubble_osqueryd ?